add collapse and tab examples to guides/contributing.rst (#4359)
OttoWinter/esphomedocs: guides/contributing.rst

    # Start web server on port 8000
    make live-html

> s

Reviewer: What's up with this? haha

Author: how did this slipped in. Let me fix

      Note that a blank line is *required* after every ``code-block`` directive.

- **Collapsible section**: To add a collapsible section, use the ``collapse`` directive:

  .. code-block:: rst

      .. collapse:: Details

          Something small enough to escape casual notice.

  .. collapse:: Details

      Something small enough to escape casual notice.

  .. code-block:: rst

      .. collapse:: A long code block

          .. code-block:: yaml

              # Sample configuration entry
              switch:
                - platform: gpio
                  name: "Relay #42"
                  pin: GPIOXX

  .. collapse:: A long code block

      .. code-block:: yaml

          # Sample configuration entry
          switch:
            - platform: gpio
              name: "Relay #42"
              pin: GPIOXX

  The ``:open:`` flag can be used to have the section open by default.

  .. code-block:: rst

      .. collapse:: Open
          :open:

          This section is open by default.

  .. collapse:: Open
      :open:

      This section is open by default.

  .. note::

>       - The ``:open:`` flag following the ``collapse`` directive has to be without a blank line.

Reviewer: ```suggestion
      - The ``:open:`` flag must immediately follow the ``collapse`` directive without a blank line between them.
```

      Note that a blank line is *required* after every ``code-block`` directive.

- **Collapsible section**: To add a collapsible section, use the ``collapse`` directive:

  .. code-block:: rst

      .. collapse:: Details

          Something small enough to escape casual notice.

  .. collapse:: Details

      Something small enough to escape casual notice.

  .. code-block:: rst

      .. collapse:: Open
          :open:

          This section is open by default.

  .. collapse:: Open
      :open:

      This section is open by default.

  .. code-block:: rst

      .. collapse:: A long code block

          .. code-block:: yaml

              # Sample configuration entry
              switch:
                - platform: gpio
                  name: "Relay #42"
                  pin: GPIOXX

  .. collapse:: A long code block

      .. code-block:: yaml

          # Sample configuration entry
          switch:
            - platform: gpio
              name: "Relay #42"
              pin: GPIOXX

  .. note::

>       - The ``:open:`` flag has to be immediately after the ``collapse`` directive without a blank line.

Reviewer: ```suggestion
      - The ``:open:`` flag must immediately follow the ``collapse`` directive without a blank line.
```

      - A blank line is *required* after every ``collapse`` directive.

> - **Tabs**: To group content into tabs, use the ``tabs`` directive. The tabs directive to define a tab set.

Reviewer: ```suggestion
- **Tabs**: To group content into tabs, use the ``tabs`` directive to define a tab set.
```
